src/main/clojure/osgi/deps.clj

```diff
@@ -0,0 +1,390 @@
+(ns osgi.deps
+  (:require [osgi.core :as osgi]
+            [clojure.java.io :as io]
+            [clojure.java.shell :refer [sh]]
+            [dorothy.core :as dot]
+            [dorothy.jvm :refer [save! show!]]))
+(comment
+  "The deps namespace is for functions that help manipulate OSGi metadata and
+  dependencies.")
+
+(defn render [graph]
+  (let [png "/tmp/render.png"]
+    (-> graph dot/dot (save! png {:format :png :layout :circo}))
+    #_(sh "open" png)
+    #_(sh "rm" png)))
+
+(defn view-by-show [graph]
+  (-> (dot/digraph graph)
+      (dot/dot)
+      (show! {:layout :neato})))
+
+(defn view-after-save [graph props]
+  (let [file "/tmp/render.svg"]
+    (-> graph dot/dot (save! file props))
+    (sh "open" file)))
+
+(defn get-deps [bundles]
+  (->> bundles
+       (mapcat (fn [bundle]
+                 (map #(-> [(first %) (:name bundle)])
+                      (get-in bundle [:headers :export-package]))))
+       (into {})))
+
+(defn package-deps []
+  (let [bundles (->> (osgi/bundle-list)
+                     (map osgi/bundle->map))
+        deps (get-deps bundles)]
+    (dot/digraph
+      [;(map #(-> [(:id %) {:label (:name %) :shape :box}]) bundles)
+       (->> bundles
+            (filter #(re-matches #"catalog-ui-search" (:name %)))
+            (mapcat
+              (fn [bundle]
+                (map #(-> [(:name bundle) (get deps (first %))])
+                     (get-in bundle [:headers :import-package])))))])))
+
+(defn service-deps []
+  (let [bundles (map osgi/bundle->map (take-last 10 (osgi/bundle-list)))]
+    (dot/digraph
+      [(map #(-> [(:id %) {:label (:name %) :shape :box}]) bundles)])))
+
+(defn
+  feature->map
+  [f] {
```

Comment:
✏️ Looking at moving all the OSGi specific stuff to the core namespace and better define the terminology / data structures that this namespace operates on.